### ISSUE TITLE
build process is broken due to deprecated gulp-cssnano dependency

### ISSUE THREAD

**Issue by jjech** [olifolkerd/tabulator#2581]
**Is your feature request related to a problem? Please describe.**
When running a new tabulator build for the first time, my first 'gulp' always fails. I was hoping that PR #2551 would solve the problem, but after merging the changes from that PR into my fork, the issue persists.

This is the error message returned from 'gulp':
```
node_modules/tabulator-tables/node_modules/gulp-cssnano/node_modules/browserslist/index.js:37
    throw new BrowserslistError(name);
    ^
BrowserslistError: Unknown browser query `dead`
    at error (node_modules/tabulator-tables/node_modules/gulp-cssnano/node_modules/browserslist/index.js:37:11)
    at node_modules/tabulator-tables/node_modules/gulp-cssnano/node_modules/browserslist/index.js:222:9
    at Array.forEach (<anonymous>)
    at browserslist (node_modules/tabulator-tables/node_modules/gulp-cssnano/node_modules/browserslist/index.js:196:13)
    at cleanBrowsersList (node_modules/tabulator-tables/node_modules/gulp-cssnano/node_modules/caniuse-api/dist/utils.js:56:59)
    at setBrowserScope (node_modules/tabulator-tables/node_modules/gulp-cssnano/node_modules/caniuse-api/dist/index.js:29:43)
    at Object.<anonymous> (node_modules/tabulator-tables/node_modules/gulp-cssnano/node_modules/caniuse-api/dist/index.js:91:1)
    at Module._compile (internal/modules/cjs/loader.js:778:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:789:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
```

The problem is that the latest released version of gulp-cssnano (2.1.3) has a dependency on cssnano 3.0.0, which in turn depends on an old version of browserslist (1.7.7) that does not contain the 'dead' browser type. This causes a build error.

**Describe the solution you'd like**
The most convenient/ideal solution would be to merge a PR to the gulp-cssnano repository updating the cssnano dependency to 4.0.0 or higher. However, someone already opened a pull request for exactly that more than a year ago (https://github.com/ben-eb/gulp-cssnano/pull/104) and it was never merged because the gulp-cssnano repository has been archived and is no longer being maintained...

**Describe alternatives you've considered**
I'm currently using the following workaround to build tabulator successfully:
1. vi tabulator-tables/node_modules/gulp-cssnano/package.json
2. change dependency cssnano from 3.0.0 to 4.0.0
3. npm install
4. gulp

**Comment by jjech** [comment]
Looks like we can fix this with npm shrinkwrap: https://nodejs.org/en/blog/npm/managing-node-js-dependencies-with-shrinkwrap/

I'll open a PR shortly --- hopefully you will be able to automerge since I will have already merged the changes from PR #2551 in my branch (they both modify package.json... running npm shrinkwrap replaces package.json with npm-shrinkwrap.json).

**Comment by jjech** [comment]
This build failure appears to be linked to something specific to my user context. Tabulator builds successfully using the gulp4 branch when outside of my user context, such as directly in a clone of the tabulator repository, or as a node_module for another node app that isn't the one I'm working on.

Closing as not enough info...
